Sprint 15

usvladis opened the pull request:
Проведена работа по 15 спринту:
1. Завершена работа над экраном добавления новой привычки.
2. Реализован слой работы с базой данных с использованием абстракций:
-добавлены классы TrackerStore, TrackerCategoryStore, TrackerRecordStore для абстрагирования Core Data;
-добавлены Entity для хранения Tracker, TrackerCategory, TrackerRecord
-в модели (файле с расширением *.xcdatamodeld) указаны взаимосвязи между Entity;
-создан один Persistent Container, который управляет тремя типами Store
-использован NSFetchedResultsController для получения обновлений от Core Data
-можно создавать привычки и нерегулярные события — и после перезапуска приложения информация отображается
-слой хранения данных на базе Core Data выполняет задачи
-приложение надёжно абстрагировано от Core Data: строчка import CoreData есть только в файлах Store.swift и месте создания NSPersistentContainer.

P.S. Это был самый сложный спринт для меня. Не потому что тут сложная тема или я где-то завис, нет. Просто я три недели отдыхал во Владивостоке и улетел как раз когда почти доделал спринт. Естественно, когда я вернулся я все забыл и пришлось учить тему заново... Да, это просто оправдание для самого себя

usvladis committed:
Доработан экран добавления привычек
Добавлена модель CoreData
Todo: Реализовать загрузку данных в и из CoreData

usvladis committed:
Трекеры загружаются из базы CoreData

TODO: Реализовать CoreData для массива TrackerRecord, то есть добавлять выполненные трекеры в память

usvladis committed:
Проведена работа по 15 спринту:
1. Завершена работа над экраном добавления новой привычки.
2. Реализован слой работы с базой данных с использованием абстракций:
-добавлены классы TrackerStore, TrackerCategoryStore, TrackerRecordStore для абстрагирования Core Data;
-добавлены Entity для хранения Tracker, TrackerCategory, TrackerRecord
-в модели (файле с расширением *.xcdatamodeld) указаны взаимосвязи между Entity;
-создан один Persistent Container, который управляет тремя типами Store
-использован NSFetchedResultsController для получения обновлений от Core Data
-можно создавать привычки и нерегулярные события — и после перезапуска приложения информация отображается
-слой хранения данных на базе Core Data выполняет задачи
-приложение надёжно абстрагировано от Core Data: строчка import CoreData есть только в файлах Store.swift и месте создания NSPersistentContainer.

usvladis committed:
Удалил и добавил файл dataModel

LizaIlinykh commented on Tracker.xcodeproj/project.pbxproj:
```diff
@@ -769,24 +769,24 @@
-			path = Tracker.xcdatamodeld;
+			currentVersion = 218DC62C2C6F997300603787 /* CoreDataModel.xcdatamodel */;
+			name = CoreDataModel.xcdatamodeld;
+			path = /Users/vladislavusacev/Documents/Tracker/CoreData/CoreDataModel.xcdatamodeld;
```
здесь создается путь к файлу на твоем компьютере. нужно, чтобы путь был такой: 
<img width="585" alt="Снимок экрана 2024-08-16 в 17 44 14" src="https://github.com/user-attachments/assets/60933111-602b-482c-a363-dd83ca9c39e9">

usvladis replied:
Я могу в проекте самостоятельно поменять путь или мне нужно снова манипуляции с удалением файла провести?

LizaIlinykh replied:
Можно, прикладываю видео
https://disk.yandex.ru/i/TK1q-UTgnDW_VQ
В конце в этом файле нажимаю cmd+f , появляется поисковая строка, ввожу туда нужную мне строку :
`path = /Users/vladislavusacev/Documents/Tracker/CoreData/CoreDataModel.xcdatamodeld;`
и удаляю не нужное в ней

usvladis committed:
Изменил путь файла project.pbxproj

usvladis committed:
Реализовал экран нерегулярного события, почему-то вообще про него забыл(
Скрывается клавиатура при создании трекера. Клавиатура скрывается при нажатии на кнопку Ввода или по нажатию на свободную область экрана
